### ISSUE TITLE
does bind() a two way binding?

### ISSUE THREAD

**Issue by Puresoft** [ocombe/angular-localForage#129]
If I bind to the same key in different controllers and one controller modify the object, this 
modification is not reflected in the other controller right?

Wouldn't it be nice to have it that way.?

**Comment by scotttrinh** [comment]
Yeah, I imagine people would expect it to work like that. A kind of "three-way" binding where changes in the scope update the DB and also update any other scope properties that are bound to the same key.

Just took a quick peek at the implementation (I didn't write it...) and it might take a little bit of work to accomplish, but seems like a good candidate for the next major version. I'd argue that this is a breaking change since people *might* be depending on the current behavior. Maybe we can make this an opt-in feature.

Stay tuned, and thanks for bringing this up!

**Comment by imwhocodes** [comment]
How it can be done with the current relase?
I tried as @Puresoft suggest, the data is update in the local $scope of the controller but the "storage" is not updated

**Comment by scotttrinh** [comment]
@imwhocodes 
You could set up the notifications like this (from the README.md):

```js
angular.module('yourModule', ['LocalForageModule'])
.config(['$localForageProvider', function($localForageProvider){
    $localForageProvider.setNotify(true, true); // itemSet, itemRemove
}]);
```

And then in your controller, you could listen for changes like:
```js
$scope.$on('LocalForageModule.setItem', function onSetItem(item) {
  // item === {key: key, newvalue: value, driver: localforage.driver};
  $scope.key = item.value;
});
```

Having said that, I would recommend *against* using `angular-localForage` as a bridge between different controllers, honestly. It's definitely a hidden coupling between controllers, and if two controllers need to share data, the typical Angular way to do this is with a service.

**Comment by scotttrinh** [comment]
Actually, that won't keep direct mutations in sync, just when using `setItem`. 🤔 Not sure that there is currently a way to do that, but I'll dive a little deeper.
